geotf/src/geodetic_converter.cc

```diff
@@ -0,0 +1,441 @@
+#include <geotf/geodetic_converter.h>
+namespace geotf {
+// Initialize frame definitions from rosparams
+void GeodeticConverter::initFromRosParam(const std::string& prefix) {
+  GeodeticConverter converter;
+  ros::NodeHandle nh;
+  if (!nh.ok() || !nh.hasParam(prefix)) {
+    ROS_WARN("[GeoTF] No GeodeticTF Transformations found.");
+    return;
+  }
+  XmlRpc::XmlRpcValue yaml_raw_data;
+  nh.getParam("/geotf", yaml_raw_data);
+
+  XmlRpc::XmlRpcValue& frame_definitions = yaml_raw_data["Frames"];
+  for (auto it = frame_definitions.begin();
+       it != frame_definitions.end(); ++it) {
+
+    const std::string frame_name = it->first;
+    XmlRpc::XmlRpcValue& xmlnode = it->second;
+
+    std::string frame_type = xmlnode["Type"];
+    if (frame_type == "EPSGCode") {
+      if (!xmlnode.hasMember("Code")) {
+        ROS_WARN_STREAM("[GeoTF] Ignoring frame " << frame_type
+                                                  << ": EPSGCode needs Code setting.");
+        continue;
+      }
+      int code = xmlnode["Code"];
+      addFrameByEPSG(frame_name, code);
+
+    } else if (frame_type == "GCSCode") {
+      if (!xmlnode.hasMember("Code")) {
+        ROS_WARN_STREAM("[GeoTF] Ignoring frame " << frame_type
+                                                  << ": GCSCode needs Code setting.");
+        continue;
+      }
+      std::string code = xmlnode["Code"];
```

Comment:
```suggestion
      auto code = xmlnode["Code"];
```

Reply:
I think here std::string could even be needed for template deduction stuff - so I leave it.